### ISSUE TITLE
Logistic regression

### ISSUE THREAD

**Issue by SaFouNaldo** [epfl-ML-2025/ms#2]
`src/methods/logistic_regression.py`
Implement multi-class logistic regression. The training will use gradient descent to learn the weights. You can search for the best learning rate and/or the best number of max iterations.
Note: if you get nan when training, it is possible that your learning rate is too high and causes the weights and/or np.exp() to diverge.
